stripe/http_client.py

```diff
@@ -105,9 +111,18 @@ def __init__(self, verify_ssl_certs=True, proxy=None):
         self._thread_local = threading.local()
 
     def request_with_retries(self, method, url, headers, post_data=None):
+        if stripe.enable_telemetry and self._last_request_metrics():
+            headers["X-Stripe-Client-Telemetry"] = json.dumps(
+                {
+                    "last_request_metrics": self._last_request_metrics().payload()
+                }
+            )
+
```

Comment:
Just now that I'm taking a second look at this, would you mind moving this into a helper method called something like `_add_telemetry_header(headers)`? Just so we have less code in the main body here.